src/value/mod.rs

```diff
@@ -167,6 +180,43 @@ pub(crate) fn mg_value_string(mg_value: *const bindings::mg_value) -> String {
     mg_string_to_string(c_str)
 }
 
+pub(crate) fn mg_value_naive_date(mg_value: *const bindings::mg_value) -> Option<NaiveDate> {
+    let c_date = unsafe { bindings::mg_value_date(mg_value) };
+    let c_delta_days = unsafe { bindings::mg_date_days(c_date) };
+    let epoch_date = NaiveDate::from_ymd(1970, 1, 1);
+    let delta_days = Duration::days(c_delta_days);
+    epoch_date.checked_add_signed(delta_days)
+}
+
+pub(crate) fn mg_value_naive_local_time(mg_value: *const bindings::mg_value) -> Option<NaiveTime> {
+    let nsec_in_sec = 1_000_000_000;
+    let c_local_time = unsafe { bindings::mg_value_local_time(mg_value) };
+    let c_nanoseconds = unsafe { bindings::mg_local_time_nanoseconds(c_local_time) };
+    // TODO(gitbuda): Check the i64 to u32 conversion.
```

Comment:
This should work https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6052c95fb0df7b20806987e85e57e88f

Reply:
Alternatively, consider the approach described on the comment below for `mg_value_naive_local_date_time`

Reply:
I did something similar to this :smile: